pysatSpaceWeather/instruments/sw_f107.py

```diff
@@ -394,72 +428,59 @@ def download(date_array, tag, inst_id, data_path):
-            if date.day != 1:
-                raise ValueError(' '.join(('The Download routine must be',
-                                           'invoked with a freq="MS"',
-                                           'option.')))
-            # download webpage
-            dstr = 'http://lasp.colorado.edu/lisird/latis/dap/'
-            dstr += 'noaa_radio_flux.json?time%3E='
-            dstr += date.strftime('%Y-%m-%d')
-            dstr += 'T00:00:00.000Z&time%3C='
-            dstr += (date + pds.DateOffset(months=1)
-                     - pds.DateOffset(days=1)).strftime('%Y-%m-%d')
-            dstr += 'T00:00:00.000Z'
-            # data returned as json
-            r = requests.get(dstr)
-            # process
-            raw_dict = json.loads(r.text)['noaa_radio_flux']
-            data = pds.DataFrame.from_dict(raw_dict['samples'])
-            if data.empty:
-                warnings.warn("no data for {:}".format(date), UserWarning)
-            else:
-                times = [dt.datetime.strptime(time, '%Y%m%d')
-                         for time in data.pop('time')]
-                data.index = times
-                # replace fill with NaNs
-                idx, = np.where(data['f107'] == -99999.0)
-                data.iloc[idx, :] = np.nan
-                # create file
-                str_date = date.strftime('%Y-%m')
-                data_file = 'f107_monthly_{:s}.txt'.format(str_date)
-                data.to_csv(os.path.join(data_path, data_file), header=True)
-
-    elif tag == 'all':
-        # download from LASP, by year
-
-        # download webpage
-        dstr = 'http://lasp.colorado.edu/lisird/latis/dap/'
-        dstr += 'noaa_radio_flux.json?time%3E='
-        dstr += dt.datetime(1947, 2, 13).strftime('%Y-%m-%d')
-        dstr += 'T00:00:00.000Z&time%3C='
-        now = dt.datetime.utcnow()
-        dstr += now.strftime('%Y-%m-%dT%H:%M:%S.000Z')
-        # data returned as json
-        r = requests.get(dstr)
-        # process
-        raw_dict = json.loads(r.text)['noaa_radio_flux']
-        data = pds.DataFrame.from_dict(raw_dict['samples'])
-        try:
-            # This is the new data format
-            times = [dt.datetime.strptime(time, '%Y%m%d')
-                     for time in data.pop('time')]
-        except ValueError:
-            # Accepts old file formats
-            times = [dt.datetime.strptime(time, '%Y %m %d')
-                     for time in data.pop('time')]
-        data.index = times
-        # replace fill with NaNs
-        idx, = np.where(data['f107'] == -99999.0)
-        data.iloc[idx, :] = np.nan
-        # create file
-        data_file = 'f107_1947_to_{:s}.txt'.format(now.strftime('%Y-%m-%d'))
-        data.to_csv(os.path.join(data_path, data_file), header=True)
-
+    if tag == 'historic':
+        # Test the date array, updating it if necessary
+        if date_array.freq != 'MS':
+            warnings.warn(''.join(['Historic F10.7 downloads should be invoked',
```

Comment:
I think I may have originally started the whole download functions only worked correct if passed the right frequency string. While working on Dst I was reminded that it filtered the passed dates to get the true range of data to download, thus removing any requirement that the user remember that Dst needs a special download setup. Made me think that we may want to remove this standard I started with the whole 'freq=something' and move towards the download function doing the right thing based upon its particular data source.

I'm not going to reviewer require this but since pysatSpaceWeather will be the first penumbra release after pysat v3 it could be the right time to discuss a few more bits of the pysat standard.

Reply:
I am in theory in support of this (the reason why I changed the error raised to a warning).  It would need to change in pysat, though, really.  Probably need to remove `freq` as a download option and let the instruments use it internally.  Thoughts?

Reply:
Yeah, your code is already there. I'm fine with changing that part of pysat.

Reply:
See https://github.com/pysat/pysat/issues/788